### DISCUSSION TITLE
Multiple Accounts

### DISCUSSION THREAD

**Discussion by scath999** [community/community#70180] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi Everyone,

Like a lot of ppl here on github, I have two accounts: one for work, one for personal projects.

I would like to push code using the appropriate accounts for the appropriate repos.

I'm not seeing a lot of help or discussion on how to do this cleanly and easily.

As such, my current solution is to use `ssh` for work projects and `https` for personal project. The latter requires me to enter my username and a token as a password _every time_.

I must be missing something - this can't be the recommended way to manage multiple accounts.

Thanks so much!

**Comment by andrej-kocijan** [comment, marked as answer]
Here is the solution I see frequently using SSH keys: [Freecodecamp](https://www.freecodecamp.org/news/manage-multiple-github-accounts-the-ssh-way-2dadc30ccaca/)
Furthermore github themselves suggest the following options: [Managing multiple accounts](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/managing-multiple-accounts)
If you are talking about two personal accounts (not enterprise) I would highly suggest merging them, I've done so recently and it is a superior expirience in my opinion. See how to merge your accounts here: [Merging Personal Accounts](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts)

**Reply by demastri** [reply]
Thanks for the pointer - Trying to merge two accounts - on the page you linked it says to attribute commits to make sure and "add the email address you used to author the commits to the account you're keeping" but when I try to add that account, I get "Error adding x@x.x, email is already in use."  Any thoughts?

**Reply by andrej-kocijan** [reply]
You have to delete the old account first or remove the email from it.

**Comment by scath999** [comment]
Thanks for the fast response - I will take a look asap!

**Reply by andrej-kocijan** [reply]
Glad to help! Feel free to mark the comment as answer if it solved your problems :)

**Reply by scath999** [reply]
Done :) Thanks again!

**Comment by ghostinhershell** [comment]
Hey @scath999 and @andrej-kocijan 👋🏾 

I gave you guys, and this question, a shout out in our Community Check-In. Check that out here: [October API & Webhooks Community Check-In 🎃](https://github.com/orgs/community/discussions/71903)

I'm following up here, however, because I thought it was important to note that [having multiple **personal** accounts is against GitHub’s terms of service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#3-account-requirements). 
- > One person or legal entity may maintain no more than one free Account (if you choose to control a machine account as well, that's fine, but it can only be used for running a machine).

Thanks so much for participating in the community!
